jobs/pkg/run/run.go

```
@@ -64,6 +66,18 @@ func Run(ctx context.Context, opts Options) error {
 		CaBundle:           certs.CaCertificate,
 	}
 
+	valid, err := kube.SecretIsValidTlsSecret(ctx, kubeClient, secretConfig)
+	if err != nil {
+		return eris.Wrapf(err, "failed validating existing secret")
```

If we fail to validate the existing tls secret, would we want to terminate early? I would have expected us to warn that we failed to validate but then just to be safe, continue with the job and generate a new one.

---

Out of curiosity, do we have any way of tracking metrics around frequency of cert gen errors/successes/skips?

---

I am leaning towards failing loudly here, with job failure, rather than silently generating new certs, because:
- If the existing cert is invalid, it is likely the user has customized the secret or
- the job has run and is trying to update the wrong secret

in either case, would rather see the user take action to delete/fix the existing "invalid" secret, rather than have our job blow it away silently. thoughts?

also, no metrics for certgen today :/

---

Yeah I agree with the benefit of failing loudly.  I didn't appreciate (understand) what would cause certgen to fail (in this case, some bad user intervention), and what occurs when the certgen job fails. I'm all for loud fails when some user intervention is required. 

I was nervous that if the existing cert was bad, then the job would just continually fail, forgetting that users could/would need to manually intervene